documentation/inventory_files.md

```diff
@@ -468,6 +468,189 @@ plugins:
+
+#### Composing project-specific inventory
+
+If your projects only share some inventory in common, or have project-specific
+configuration for targets, you can use the `yaml` plugin to compose an inventory
+for each project.
+
+For example, you might have two projects where both projects run on a set of web
+servers, but one of the projects also runs on a set of database servers. Both
+projects also specify different login credentials.
+
+To compose an inventory for each project, you can save each set of targets to a
+separate shared inventory, and then reference them as needed using the `yaml`
+plugin in each project's inventory file.
+
+Save the set of web servers to
+`~/.puppetlabs/etc/bolt/server_inventory.yaml`:
+
+```yaml
+- server-1.example.org
+- server-2.example.org
+- server-3.example.org
+```
+
+Save the set of database servers to
+`~/.puppetlabs/etc/bolt/database_inventory.yaml`:
+
+```yaml
+- db-1.example.org
+- db-2.example.org
+- db-3.example.org
+```
+
+Save an inventory file to each project that loads the necessary targets:
+
+```yaml
+# ~/projects/servers/inventory.yaml
+groups:
+  - name: servers
+    targets:
+      _plugin: yaml
+      filepath: ~/.puppetlabs/etc/bolt/server_inventory.yaml
+config:
+  ssh:
+    user: admin
+    password: admin
+```
+
+```yaml
+# ~/projects/webapp/inventory.yaml
+groups:
+  - name: servers
+    targets:
+      _plugin: yaml
+      filepath: ~/.puppetlabs/etc/bolt/server_inventory.yaml
+  - name: databases
+    targets:
+      _plugin: yaml
+      filepath: ~/.puppetlabs/etc/bolt/database_inventory.yaml
+config:
+  ssh:
+    user: puppet
+    password: Bolt!
+```
+
+#### Specifying target defaults
+
+Shared inventory files allow you to specify default configuration and data for
+targets that can be overridden with project-specific configuration and data.
+If a target is listed multiple times in an inventory file, then the first
+instance of the target in the inventory file [takes precedence](#precedence)
+over subsequent instances of the target.
+
+For example, the following inventory file includes the `example.org` target. The
+target is defined in the project's inventory file and is also defined in a
+shared inventory file with some default configuration and data.
+
+```yaml
+# ~/projects/example/inventory.yaml
+groups:
+  - name: servers
+    targets:
+      - uri: example.org
+        name: example
+        config:
+          ssh:
+            password: Bolt!
+  - name: _defaults
+    targets:
+      _plugin: yaml
+      filepath: ~/.puppetlabs/etc/bolt/shared_inventory.yaml
+```
+
+```yaml
+# ~/.puppetlabs/etc/bolt/shared_inventory.yaml
+- uri: example.org
+  name: example
+  config:
+    ssh:
+      user: admin
+      password: Puppet!
+  vars:
+    role: server
+```
+
+When Bolt loads this inventory, the target's password will be set to `Bolt!`
+because the project-specific target definition comes before the target
+definition in the shared inventory.
```

Comment:
```suggestion
Bolt uses the password `Bolt!` because the project-specific target
definition comes before the target definition in the shared inventory.
```